analyses/molecular-subtyping-CRANIO/00-craniopharyngiomas-molecular-subtype.Rmd

```diff
@@ -0,0 +1,173 @@
+if (!dir.exists(results_dir)) {
+  dir.create(results_dir)
+}
+
+# Outputs
+output_lesions <- file.path(results_dir, "CRANIO_defining_lesions.tsv")
+output_subtype <- file.path(results_dir, "CRANIO_molecular_subtype.tsv")
+```
+
+```{r load-inputs-please-wait}
+# Inputs
+## Read in metadata
+metadata <-
+  read_tsv(file.path(root_dir, "data", "pbta-histologies.tsv")) %>%
+  filter(pathology_diagnosis == "Craniopharyngioma")
+## Read in snv consensus mutation data
+snv_df <-
+  read_tsv(file.path(root_dir, "data", "pbta-snv-consensus-mutation.maf.tsv.gz"))
+```
+
+# Process Data
+
+## Collecting Tumor Sample Barcodes with Variants of Interest
+
+Collect Tumor_Sample_Barcodes with BRAF V600E mutations.
+
+```{r braf-tsb}
+braf_v600e_tsb <-
+  snv_df %>%
+  filter(Hugo_Symbol == "BRAF" & HGVSp_Short == "p.V600E") %>%
+  pull(Tumor_Sample_Barcode)
+```
+
+Collect Tumor_Sample_Barcodes with CTNNB1 mutation on exon 3.
+
+```{r ctnnb1-tsb}
+ctnnb1_e3_tsb <-
+  snv_df %>%
+  filter(Hugo_Symbol == "CTNNB1" & Exon_Number == "3/15") %>%
+  pull(Tumor_Sample_Barcode)
+```
+
+## Defining Lesion Table
+
+Annotate the metadata table where the Kids_First_Biospecimen_ID matches either of the Tumor_Sample_Barcodes (TSBs) collected above
+Select the columns from the table that the PIs have requested
+
+```{r snv-lesion-table}
+snv_lesion_df <-
+  metadata %>%
+  mutate(CTNNB1.Exon3 = Kids_First_Biospecimen_ID %in% ctnnb1_e3_tsb) %>%
+  mutate(BRAF.V600E = Kids_First_Biospecimen_ID %in% braf_v600e_tsb) %>%
+  select(Kids_First_Participant_ID,
+         sample_id,
+         Kids_First_Biospecimen_ID,
+         age_at_diagnosis_days,
+         CTNNB1.Exon3,
+         BRAF.V600E)
+```
+
+## Defining Molecular Subtype Table
+
+Make a frame with DNA and RNA samples per patient
+
+```{r allmeta-table}
+dnameta <-
+  metadata %>%
+  filter(experimental_strategy %in% c("WGS","WXS")) %>%
+  select(Kids_First_Participant_ID,sample_id,Kids_First_Biospecimen_ID)
+
+rnameta <-
+  metadata %>%
+  filter(experimental_strategy == "RNA-Seq") %>%
+  select(Kids_First_Participant_ID,sample_id,Kids_First_Biospecimen_ID)
+
+allmeta <-
+  full_join(dnameta,
+            rnameta,
+            by=c("Kids_First_Participant_ID","sample_id"),
+            suffix = c("_DNA", "_RNA"))
+```
+
+Append molecular subtypes defined as follows:
+1. `CRANIO, PAP`: patient has BRAF.V600E mutation and is older than the age_of_adulthood (40 years in this case)
+1. `CRANIO, ADAM`:patient has CTNNB1.Exon3 mutation and is younger than the age_of_adulthood (40 years in this case)
+1. `CRANIO, ADAM;CRANIO, PAP` Both of the above (this is actually impossible given the restriction on age)
+1. `CRANIO, To be classified` None of the above
+Select the columns from the table that the PIs have requested
+
+```{r meta-subtype-table}
+# Dispute this all you like
+age_of_adulthood <- 40 * 365.25
+
+snv_subtype <-
+  snv_lesion_df %>%
+  mutate(molecular_subtype =
```

Comment:
To save yourself from repeating `as.numeric()` a bunch of times, as well as `>= age_of_adulthood` you can just make one age boolean variable. 

`older_than_40` doesn't need to be the variable name, but whatever you think works. 
```suggestion
  mutate(older_than_40 = as.numeric(age_at_diagnosis_days) >= age_of_adulthood, 
```

Reply:
Done